umbral/umbral.py

```diff
@@ -115,98 +116,135 @@ def __bytes__(self):
 
 
 class Capsule(object):
-    def __init__(self, point_eph_e, point_eph_v, bn_sig):
-        self.point_eph_e = point_eph_e
-        self.point_eph_v = point_eph_v
-        self.bn_sig = bn_sig
+    def __init__(self,
+                 point_eph_e=None,
+                 point_eph_v=None,
+                 bn_sig=None,
+                 e_prime=None,
+                 v_prime=None,
+                 noninteractive_point=None):
+
+        if not point_eph_e and not e_prime:
+            raise ValueError(
+                "Can't make a Capsule from nothing.  Pass either Alice's data (ie, point_eph_e) or Bob's (e_prime). \
+                Passing both is also fine.")
+
+        self._point_eph_e = point_eph_e
+        self._point_eph_v = point_eph_v
+        self._bn_sig = bn_sig
+
+        self._point_eph_e_prime = e_prime
+        self._point_eph_v_prime = v_prime
+        self._point_noninteractive = noninteractive_point
 
         self.cfrags = {}
+        self._contents = None
 
-    @staticmethod
-    def from_bytes(data: bytes, curve: ec.EllipticCurve):
+    @classmethod
+    def from_original_bytes(cls, data: bytes, curve: ec.EllipticCurve):
         """
         Instantiates a Capsule object from the serialized data.
         """
         eph_e = Point.from_bytes(data[0:33], curve)
         eph_v = Point.from_bytes(data[33:66], curve)
         sig = BigNum.from_bytes(data[66:98], curve)
 
-        return Capsule(eph_e, eph_v, sig)
+        return cls(eph_e, eph_v, sig)
+
+    @classmethod
+    def from_reconstructed_bytes(cls, data: bytes, curve: ec.EllipticCurve):
```

Comment:
Please rename to `from_bytes` for uniformity.

Reply:
Well, which one is `from_bytes`?  This or `from_original_bytes`?

Reply:
I would remove both and leave in `from_bytes`, and offer a flag for `is_reconstructed=False`. When `True`, reconstruct the capsule.

Reply:
Ahh yes that sounds good.